apps/managedidentity/managedidentity.go

```diff
@@ -178,8 +199,47 @@ func createIMDSAuthRequest(ctx context.Context, id ID, resource string, claims s
 	return req, nil
 }
 
+// Contains checks if the element is present in the list.
+func contains[T comparable](list []T, element T) bool {
+	for _, v := range list {
+		if v == element {
+			return true
+		}
+	}
+	return false
+}
+
+// retry performs an HTTP request with retries based on the provided options.
+func retry(maxRetries int, c ops.HTTPClient, req *http.Request) (*http.Response, error) {
+	var resp *http.Response
+	var err error
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		tryCtx, tryCancel := context.WithTimeout(req.Context(), time.Second*15)
+		defer tryCancel()
+		cloneReq := req.Clone(tryCtx)
+		resp, err = c.Do(cloneReq)
+		if err == nil && !contains(retryStatusCodes, resp.StatusCode) {
+			return resp, nil
+		}
+		if attempt == maxRetries-1 {
+			return resp, err
+		}
+		if resp != nil && resp.Body != nil {
+			io.Copy(io.Discard, resp.Body)
```

Comment:
Why?

Reply:
Discussed Offline about this memory leak